### DISCUSSION TITLE
How to set task input in multi-level folder structure ?

### DISCUSSION THREAD

**Discussion by thibthib** [vercel/turborepo#5947] [Help]
Hello 👋

We're heavily using turbo at @DataDog and are facing an issue in our configuration:

We have a package structure that has several levels:
```
- packages
  - lib
    - package-1
    - package-2
    -  ...
  - apps
    - my-app
      - package-3
      - package-4
      - ...
```

We'd like our tools folders to invalidate caches whenever changed. For example, a change in our `packages/lib/eslint` package should invalidate the cache of our `lint` task. So far we've been doing this with with `globalDependencies` but this comes with two main drawbacks:
- it invalidates all our tasks caches when it should only do it for the `lint` one
- _all_ files are taking into account for the hash, even git ignored ones, which forces us to be really specific and prevents us to use glob patterns

I know it should be possible to specify [inputs at the task level](https://turbo.build/repo/docs/reference/configuration#inputs) via the configuration, but with a single relative path it does not work with our folder structure. 

What can I do to improve this situation ?

**Comment by anthonyshew** [comment]
Could you make a reproduction with the steps to produce the behavior you want vs. don't want? Trying to make sure I understand your ask. 🙏

**Comment by mehulkar** [comment]
I'd need a reproduction to be sure, but it sounds like your `packages/lib/eslint` package is installed in the root package.json, rather than in each of the pacakges its used in. If you did the latter, turbo would be able to track changes to this dependency and invalidate caches of `lint` based on that.

**Comment by thibthib** [comment]
👋 Sorry for the late answer, I did not have the time on working on a reproduction.

Indeed, another solution would be to add our `lint` package as a dependency of every package. But we stumbled upon another issue doing it: circular dependencies. Let's imagine that we also have a `packages/lib/test` package, and that we want to lint that package while also testing the `lint` package. Having them global helps us solve this circular dependency.

**Reply by mehulkar** [reply]
I'm not sure I can wrap my head around this without seeing an example, but it _should_ be possible to avoid circular dependencies with the right structure. Happy to close this down if a reproduction isn't possible!
